patches/server/0738-Fix-entities-glitching-through-blocks-pushed-by-pist.patch

```diff
@@ -0,0 +1,80 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Peter Svensson <petersv@psv.nu>
+Date: Mon, 2 Aug 2021 12:06:05 +0200
+Subject: [PATCH] Fix entities glitching through blocks pushed by pistons
+ (#5189)
+
+Fix #5189 where entities in slime blocks are unreliable.
+
+Add a general mechanism where the entity knows how many ticks into
+the future that it has to be fully ticked. ActivationRange respects
+that and does not cull any of those ticks to the entity.
+
+diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
+index 896d892237b29eb404398db07264eb6f04786754..b16b51427b0d397b3e1b50d475bdf6ef5cd4e660 100644
+--- a/src/main/java/net/minecraft/world/entity/Entity.java
++++ b/src/main/java/net/minecraft/world/entity/Entity.java
+@@ -690,6 +690,11 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, n
+ 
+         this.firstTick = false;
+         this.level.getProfiler().pop();
++        // Paper start - track when all ticks are required for the entity, e.g. after a piston push
++        if (this.isTickRequired()) {
++           this.remainingExactTicks -= 1;
++        }
++        // Paper end
+     }
+ 
+     public void setSharedFlagOnFire(boolean onFire) {
+@@ -3992,4 +3997,16 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, n
+         return ((ServerChunkCache) level.getChunkSource()).isPositionTicking(this);
+     }
+     // Paper end
++ 
++    // Paper start - track when all ticks are required for the entity, e.g. after a piston push
++    private int remainingExactTicks = 0;
++
++    public void requestExactTicks(int ticks) {
++        this.remainingExactTicks = Math.max(ticks, this.remainingExactTicks);
++    }
++
++    public boolean isTickRequired() {
++        return (this.remainingExactTicks > 0);
++    }
++    // Peter end
+ }
+diff --git a/src/main/java/net/minecraft/world/level/block/piston/PistonMovingBlockEntity.java b/src/main/java/net/minecraft/world/level/block/piston/PistonMovingBlockEntity.java
+index 87bedba9ab495edcce289c6665271d92b7165944..2434e82eeb8deb9c3a4820fe34840cf5c97e7610 100644
+--- a/src/main/java/net/minecraft/world/level/block/piston/PistonMovingBlockEntity.java
++++ b/src/main/java/net/minecraft/world/level/block/piston/PistonMovingBlockEntity.java
+@@ -140,6 +140,7 @@ public class PistonMovingBlockEntity extends BlockEntity {
+                                 }
+ 
+                                 entity.setDeltaMovement(e, g, h);
++                                entity.requestExactTicks(2);    // Paper - track when all ticks are required for the entity, e.g. after a piston push
```

Comment:
This and the below hunk need reformatting

Reply:
Will you handle that as a part of the merge, or do you want me to fix it and resubmit?

Reply:
Either or; your approval only gets removed if a larger change comes

Reply:
I'll be away for the rest of the day.  If it is not already merged when I get back I can update the formatting.

Reply:
I am not sure what formatting is requested. Should I switch to the multi-line format for the // Paper marker, or was it something else you had in mind?

Reply:
i think she means the 4 spaces before the comment. everything else in paper just uses 1 (`; // Paper - `)

Reply:
Updated. let me knwo if I got this right.